### ISSUE TITLE
Clickhouse Code Suggestion (autocomplete) is broken after DBeaver update >=24.1.4

### ISSUE THREAD

**Issue by gulivan** [dbeaver/dbeaver#35708]
### Description

After updating DBeaver CE to 2.1.4, we've noticed that code completion isn't working for Clickhouse.
We've played with code completion settings but it didn't help.
Then we've tried to downgrade the version — and code completion works again.

### DBeaver Version

CE >=24.1.4

### Operating System

MacOS Sonoma 14.5

### Database and driver

1. Clickhouse, 23.8.16.40
2. Official ClickHouse driver (server version 21.3+)
com.clickhouse:clickhouse-jdbc:RELEASE 0.6.1
also the same with:
com.clickhouse:clickhouse-jdbc:RELEASE 0.6.5

### Steps to reproduce

1. Download Dbeaver 24.2.1 (latest) <=, >=24.1.4
2. Connect to Clickhouse, install proposed driver

24.1.3 (latest working)
Code auto completion works as intended
<img width="1067" alt="image" src="https://github.com/user-attachments/assets/72af0bd6-5d05-424f-95bc-32e303e40808">

24.2.1 (latest):
Code completion isn't working anymore

![image](https://github.com/user-attachments/assets/24af5f5a-de36-4ba9-9d85-0f62e04277fe)

24.1.4 (first version when code completion stopped working):

<img width="754" alt="image" src="https://github.com/user-attachments/assets/aa4dd6d2-c1ab-455f-9e1c-be044e9b0d59">

### Additional context

Code completion still works with trino, postgresql.

**Comment by E1izabeth** [comment]
Unfortunately, I can't reproduce it. Could you send the screenshot of your Code Completion settings?
![image](https://github.com/user-attachments/assets/993ff606-7892-4b50-a64e-d228c474a513)

**Comment by gulivan** [comment]
Hi @E1izabeth , thank you for looking into this.

With [dbeaver-ce-24.1.4-macos-aarch64.dmg](https://github.com/dbeaver/dbeaver/releases/download/24.1.4/dbeaver-ce-24.1.4-macos-aarch64.dmg):

Auto-completion isn't working
<img width="423" alt="image" src="https://github.com/user-attachments/assets/f51f256b-cff3-4614-937a-23ae19bf74b6">



I've tried the same settings as in your screenshot but the issue persist:
<img width="1307" alt="image" src="https://github.com/user-attachments/assets/f4bfdd36-5d89-47a5-93d6-3b9f99eda7a1">


Downgraded to 24.1.3 and it works again:

<img width="767" alt="image" src="https://github.com/user-attachments/assets/f9a58eac-37b6-49a7-a32d-9e0c36c49304">

Code completion settings for 24.1.3:

<img width="1343" alt="image" src="https://github.com/user-attachments/assets/1141e653-c134-48cd-b03d-7e5124d494ad">

I could also try to reproduce this with Windows OS later.

**Comment by E1izabeth** [comment]
Thank you for checking. I believe it doesn't depend on the OS

**Comment by HocKu7** [comment]
@gulivan Hi. Could you let me know if you managed to reproduce it on Windows?

**Comment by gulivan** [comment]
@HocKu7 Hi! Yeah, here are the results.

> TLDR autocompletion works after background parsing is done on newer versions (24.1.4, 24.2.4). In the specific version (24.1.3) the autocompletion worked without any background parsing just right out the box.

24.2.4 (Background parsing is in progress, autocompletion isn't working):
<img width="973" alt="SCR-20241107-nyep-2" src="https://github.com/user-attachments/assets/2a175000-b351-4842-8014-34637cbbcc9f">

24.2.4 (Background parsing is done, autocompletion is working):
<img width="883" alt="image" src="https://github.com/user-attachments/assets/af2bfdc2-04c4-47b2-a93f-94515eff3e4c">

I could only guess that in 24.1.4 some alterations were implemented in the parsing mechanism, cause with 24.1.3 autocompletion works just after the launch while it takes some time for newer versions (roughly 10+ minutes).

I also did some test with Windows and local clickhouse on 24.1.3, 24.1.4, autocompletion worked without any issues:
![telegram-cloud-document-2-5287697476927642183](https://github.com/user-attachments/assets/1c79aaca-522d-40d6-a282-036e4c92dada)
![telegram-cloud-document-2-5287697476927642184](https://github.com/user-attachments/assets/03832df2-e838-4b79-bd7b-341dc4d6e73c)

Maybe it also depends on the overall tables quantity, since in our prod database we've got 500+ tables.


I think it would be safe to close the ticket for now. Thank you for your attention
